src/link-manager.js

```diff
@@ -0,0 +1,12 @@
+/**
+ * @fileoverview Link manager for controlling the link query.
+ */
+
+/** @const */
+genotet.linkManager = {};
+
+/**
+ * Array that stores the preset view names of all the views.
+ */
+/** @type {!Array<string>} */
+genotet.linkManager.presetStatus = [];
```

Comment:
I am arguing that link manager does not need know whether a view is preset. There is no such isPreset property required on the views. Whether a view is preset is simply depending on whether it has associated (action, response). This is implicitly defined.